services/src/test/scala/cromwell/services/healthmonitor/HealthMonitorServiceActorSpec.scala

```diff
@@ -1,160 +1,210 @@
 package cromwell.services.healthmonitor
 
-import akka.actor.Props
+import akka.actor.{ActorRef, Props}
 import akka.pattern.ask
 import akka.util.Timeout
-import com.typesafe.config.ConfigFactory
+import com.typesafe.config.{Config, ConfigFactory}
 import cromwell.core.TestKitSuite
-import cromwell.services.healthmonitor.HealthMonitorServiceActor.{GetCurrentStatus, MonitoredSubsystem, StatusCheckResponse, SubsystemStatus, _}
-import cromwell.services.healthmonitor.HealthMonitorServiceActorSpec.{MockHealthMonitorServiceActor, _}
-import org.scalatest.AsyncFlatSpecLike
+import cromwell.services.healthmonitor.HealthMonitorServiceActor._
+import cromwell.services.healthmonitor.HealthMonitorServiceActorSpec._
+import org.scalatest.{Assertion, FlatSpecLike}
+import org.scalatest.concurrent.Eventually
 
-import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.duration._
+import scala.concurrent.{Await, Future}
 import scala.language.postfixOps
 
-class HealthMonitorServiceActorSpec extends TestKitSuite with AsyncFlatSpecLike {
+class HealthMonitorServiceActorSpec extends TestKitSuite with FlatSpecLike with Eventually {
   implicit val timeout = Timeout(5.seconds)
 
+  override val patienceConfig = PatienceConfig(timeout = scaled(20 seconds), interval = scaled(1 second))
+  implicit val patience = patienceConfig
+
+  private def eventualStatus(actorRef: ActorRef, ok: Boolean, systems: (MonitoredSubsystem, SubsystemStatus)*): Assertion = {
+    case class NameAndStatus(name: String, status: SubsystemStatus)
+
+    eventually {
+      val resp = Await.result(actorRef.ask(GetCurrentStatus).mapTo[StatusCheckResponse], Duration.Inf)
+      assert(resp.ok == ok)
+
+      val actual = resp.systems.toList.map(NameAndStatus.tupled).sortBy(_.name)
+      val expected = systems.map { case (m, s) => m.name -> s } map { NameAndStatus.tupled } sortBy(_.name)
+      assert(actual.length == expected.length)
+
+      actual.zip(expected) map { case (a, e) =>
+        assert(a.name == e.name)
+        assert(a.status.ok == e.status.ok)
+        val isEmpty = a.status.messages.isEmpty && e.status.messages.isEmpty
+        lazy val actualPrefixedByExpected = for {
+          am <- a.status.messages.toList.flatten.headOption
+          em <- e.status.messages.toList.flatten.headOption
+        } yield am.startsWith(em)
+        assert(isEmpty || actualPrefixedByExpected.contains(true))
+      } head
```

Comment:
OOC why do you need the `head` here ?

Reply:
`eventually` wants an `Assertion` result not a `List[Assertion]`